src/odemis/acq/stream/_sync.py

```diff
@@ -2072,10 +2102,35 @@ def _runAcquisition(self, future):
             for s in self._streams:
                 s._unlinkHwVAs()
             self._acq_data = [[] for _ in self._streams]  # regain a bit of memory
+            self._live_data = [[] for _ in self._streams]
+            self._streams[0].raw = []
+            self._streams[0].image.value = None
             self._dc_estimator = None
             self._current_future = None
             self._acq_done.set()
 
+    def _updateImage(self):
+        """
+        Function called by image update thread which handles updating the overlay of the SEM live update image
+        """
+        # Display only the SEM image taken in the last polarization and convert to RGB + display
+        for stream in range(2):
```

Comment:
I'd suggest to write it like `for i, stream in enumerate(self._stream)`.
The, you can replace the `stream` -> `i`, and `self.streams[stream]` -> `stream`.